src/main/java/com/esri/samples/na/find_service_areas_for_multiple_facilities/FindServiceAreasForMultipleFacilitiesSample.java

```diff
@@ -0,0 +1,230 @@
+
+    // create a graphics overlay for displaying service areas
+    GraphicsOverlay serviceAreasGraphicsOverlay = new GraphicsOverlay();
+
+    // add the graphics overlay to the map view
+    mapView.getGraphicsOverlays().add(serviceAreasGraphicsOverlay);
+
+    // create fill symbols for rendering the result
+    ArrayList<SimpleFillSymbol> fillSymbols = new ArrayList<>();
+    fillSymbols.add(new SimpleFillSymbol(SimpleFillSymbol.Style.SOLID, 0x66FFA500, null));
+    fillSymbols.add(new SimpleFillSymbol(SimpleFillSymbol.Style.SOLID, 0x66FF0000, null));
+
+    // create a feature table of facilities using a FeatureServer
+    ArcGISFeatureTable facilitiesTable = new ServiceFeatureTable("https://services2.arcgis.com/ZQgQTuoyBrtmoGdP/ArcGIS/rest/services/San_Diego_Facilities/FeatureServer/0");
+    // create a feature layer from the table
+    FeatureLayer facilitiesFeatureLayer = new FeatureLayer(facilitiesTable);
+
+    // create a symbol used to display the facilities
+    PictureMarkerSymbol facilitySymbol = new PictureMarkerSymbol("http://static.arcgis.com/images/Symbols/SafetyHealth/Hospital.png");
+    facilitySymbol.setHeight(25);
+    facilitySymbol.setWidth(25);
+    // set the renderer of the facilities feature layer to use the facilities symbol
+    facilitiesFeatureLayer.setRenderer(new SimpleRenderer(facilitySymbol));
+
+    // add the facilities feature layer to the map
+    map.getOperationalLayers().add(facilitiesFeatureLayer);
+
+    // wait for the facilities feature layer to load
+    facilitiesFeatureLayer.addDoneLoadingListener(() -> {
+      if (facilitiesFeatureLayer.getLoadStatus() == LoadStatus.LOADED) {
+
+        // zoom to the extent of the feature layer
+        mapView.setViewpointGeometryAsync(facilitiesFeatureLayer.getFullExtent(), 130);
+
+        // enable the find service areas button when the draw status is completed for the first time
+        mapView.addDrawStatusChangedListener(new DrawStatusChangedListener() {
+          @Override
+          public void drawStatusChanged(DrawStatusChangedEvent drawStatusChangedEvent) {
+            if (drawStatusChangedEvent.getDrawStatus() == DrawStatus.COMPLETED) {
+              // enable the 'find service areas' button
+              findServiceAreasButton.setDisable(false);
+              mapView.removeDrawStatusChangedListener(this);
+            }
+          }
+        });
+
+        // determine the service areas and display them when the button is clicked
+        findServiceAreasButton.setOnAction(event -> {
+
+          // disable the button
+          findServiceAreasButton.setDisable(true);
+
+          // show the progress indicator
+          progressIndicator.setVisible(true);
+
+          // create a service area task from URL
+          ServiceAreaTask serviceAreaTask = new ServiceAreaTask("https://sampleserver6.arcgisonline.com/arcgis/rest/services/NetworkAnalysis/SanDiego/NAServer/ServiceArea");
+          serviceAreaTask.loadAsync();
```

Comment:
its not necessary to call loadAsync here. Afaik this task follows the pattern of other tasks which don't need loaded to get the default params async. It is then loaded in the background when you run it. I tried without loading and the sample worked for me. If you do need to load, then you'd expect some code that takes action on done loading.

Reply:
Thanks - I've been adding this useless code on many occasions now. I should make it a habit to leave it out unless necessary!